### PR TITLE
[th2-5267] Shutdown processor thread in MessagePersistor.close method

### PR DESCRIPTION
**Pull request by Nikita-Smirnov-Exactpro**
* Split ErrorCollector to interface and implementation
* Publish artifact to sonatype for reusing in other projects

### CODE REVIEW

File: src/test/java/com/exactpro/th2/mstore/TestErrorCollector.java [hunk 1]
```diff
@@ -51,8 +51,8 @@
 @SuppressWarnings("DynamicRegexReplaceableByCompiledPattern")
 class TestErrorCollector {
 
-    private static final long PERIOD = RandomUtils.nextLong(0, Long.MAX_VALUE);
-    private static final TimeUnit TIME_UNIT = TimeUnit.values()[RandomUtils.nextInt(0, TimeUnit.values().length)] ;
+    private static final long PERIOD = RandomUtils.secure().randomLong(0, Long.MAX_VALUE);
+    private static final TimeUnit TIME_UNIT = TimeUnit.values()[RandomUtils.secure().randomInt(0, TimeUnit.values().length)] ;
```
[the remark this answers is not in the document]
Comment (Nikita-Smirnov-Exactpro) [reply src/test/java/com/exactpro/th2/mstore/TestErrorCollector.java:55]: It doesn't meter in this case, I took the first one that came to hand

File: src/main/java/com/exactpro/th2/mstore/MessagePersistor.java [hunk 2]
```diff
@@ -155,6 +163,17 @@ public void close () {
             stopped = true;
             futures.awaitRemaining();
             LOGGER.info("All waiting futures are completed");
+
+            Thread thread = persistor.get();
+            if (thread != null && thread.isAlive()) {
+                thread.interrupt();
+                thread.join(5000);
```
[the remark this answers is not in the document]
Comment (Nikita-Smirnov-Exactpro) [reply src/main/java/com/exactpro/th2/mstore/MessagePersistor.java:170]: Maybe reuse TerminationTimeout for that? Currently it is used in AbstractMessageProcessor.close method to await drain  executor termination

[the remark this answers is not in the document]
Comment (OptimumCode) [reply src/main/java/com/exactpro/th2/mstore/MessagePersistor.java]: We can I think, since it is more or less related to the resistor as well

**Review by OptimumCode — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM gradle:8.11.1-jdk11 AS build
+FROM gradle:8.11.1-jdk21 AS build
```
[the remark this answers is not in the document]
Comment (Nikita-Smirnov-Exactpro) [reply Dockerfile:1]: I would like to be sure that our component can be run on java 11. Let's update a bit later

File: src/test/java/com/exactpro/th2/mstore/TestErrorCollector.java [hunk 4]
```diff
@@ -51,8 +51,8 @@
 @SuppressWarnings("DynamicRegexReplaceableByCompiledPattern")
 class TestErrorCollector {
 
-    private static final long PERIOD = RandomUtils.nextLong(0, Long.MAX_VALUE);
-    private static final TimeUnit TIME_UNIT = TimeUnit.values()[RandomUtils.nextInt(0, TimeUnit.values().length)] ;
+    private static final long PERIOD = RandomUtils.insecure().randomLong(0, Long.MAX_VALUE);
+    private static final TimeUnit TIME_UNIT = TimeUnit.values()[RandomUtils.insecure().randomInt(0, TimeUnit.values().length)] ;
```
[the remark this answers is not in the document]
Comment (Nikita-Smirnov-Exactpro) [reply src/test/java/com/exactpro/th2/mstore/TestErrorCollector.java:55]: Looks like to check different values for scheduled periodic task.

[the remark this answers is not in the document]
Comment (OptimumCode) [reply src/test/java/com/exactpro/th2/mstore/TestErrorCollector.java:55]: And what values do we check? If we have no idea which values we check probably we should not use random values

[the remark this answers is not in the document]
Comment (OptimumCode) [reply src/test/java/com/exactpro/th2/mstore/TestErrorCollector.java:55]: We can use parameterized tests to check corner values if we need that

File: src/test/java/com/exactpro/th2/mstore/TestErrorCollector.java [hunk 5]
```diff
@@ -88,6 +88,21 @@ void afterEach() {
         verifyNoMoreInteractions(eventRouter);
     }
 
+    @ParameterizedTest
+    @CsvSource({
+            "60000,MICROSECONDS",
+            "60,SECONDS",
+            "1,MINUTES",
+    })
+    void testDrainTaskParameters(String period, String timeUnit) throws Exception {
```
Comment (OptimumCode) [inline src/test/java/com/exactpro/th2/mstore/TestErrorCollector.java:97]: note: I think JUnit is smart enough to convert value to a long - have you tried using `long period` as a parameter instead of a string?

**Review by OptimumCode — APPROVED** [review]

File: src/test/java/com/exactpro/th2/mstore/TestErrorCollector.java (hunk 5 above)
Comment (Nikita-Smirnov-Exactpro) [reply src/test/java/com/exactpro/th2/mstore/TestErrorCollector.java:97]: Good point, some version ago JUnit couldn't do that

**Review by OptimumCode — APPROVED** [review]
